pkg/lib/packagekit.es6

```diff
@@ -264,3 +264,83 @@ export function watchRedHatSubscription(callback) {
+ * actually install them.
+ *
+ * check_optional_packages resolves to either null when no packages
+ * need to be installed, or an object that can be passed to
+ * install_optional_packages.
+ *
+ * In the future, that object might contain more information about
+ * how many packages need to be installed, and how much space that
+ * will take, but, PackageKit...
+ */
+
+export function check_optional_packages(names, progress_cb) {
+    if (names.length == 0)
+        return cockpit.resolve(null);
+
+    function refresh() {
+        return cancellableTransaction("RefreshCache", [ false ], progress_cb);
+    }
+
+    function resolve() {
+        var ids = [ ];
+        return cancellableTransaction("Resolve",
+                                      [ Enum.FILTER_ARCH |
+                                        Enum.FILTER_NOT_SOURCE |
+                                        Enum.FILTER_NEWEST,
+                                        names
+                                      ],
+                                      progress_cb,
+                                      {
+                                          Package: (info, package_id) => {
+                                              console.log("P", info, package_id);
+                                              var repos = package_id.split(";")[3].split(":");
+                                              if (repos.indexOf("installed") == -1)
+                                                  ids.push(package_id);
+                                          },
+                                      }).then(() => {
+                                          if (ids.length == 0)
+                                              return null;
+                                          else
+                                              return { pkg_ids: ids };
+                                      });
+    }
+
+    function get_deps(data) {
+        if (!data)
+            return data;
+
+        console.log(data);
+
+        return cancellableTransaction("DependsOn",
+                                      [ Enum.FILTER_ARCH |
+                                        Enum.FILTER_NOT_SOURCE |
+                                        Enum.FILTER_NEWEST,
+                                        data.pkg_ids,
+                                        true
+                                      ],
+                                      progress_cb,
+                                      {
+                                          Package: (info, package_id) => {
+                                              console.log("D", info, package_id);
+                                          },
+                                      }).then(() => {
+                                          return data;
+                                      });
+    }
+
+    return refresh().then(resolve).then(get_deps);
+}
+
+export function install_optional_packages(data, progress_cb) {
```

Comment:
This function does almost nothing on top of InstallPackages, so it feels a bit like noise. But I have no strong opinion if you want to keep this wrapper. But then please call it `install_packages()` and give it its own documentation comment.